### PR TITLE
fix(dropdown): dropdown wrapper no longer overflows inside a panel

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #3048 

## Summary

This fixes  `overflow` caused by dropdown wrapper when wrapped inside a `panel`  with `filter` in a `pick-list`

### CODE REVIEW

File: src/components/calcite-dropdown/calcite-dropdown.tsx [hunk 1]
```diff
@@ -161,7 +161,9 @@ export class CalciteDropdown {
             onTransitionEnd={this.transitionEnd}
             ref={this.setScrollerEl}
```
Comment (driskull) [inline src/components/calcite-dropdown/calcite-dropdown.tsx:162]: This seems to work for me, I don't think we need another div element.

```
<div
            class={{
              ["calcite-dropdown-content"]: true,
              [PopperCSS.animation]: true,
              [PopperCSS.animationActive]: active
            }}
            hidden={!active}
            onTransitionEnd={this.transitionEnd}
            ref={this.setScrollerEl}
          >
```

Comment (anveshmekala) [reply src/components/calcite-dropdown/calcite-dropdown.tsx:162]: The issue with this approach is that `calciteDropOpen` event isn't emitted as the element is hidden.

Comment (driskull) [reply src/components/calcite-dropdown/calcite-dropdown.tsx:162]: Gotcha. makes sense

**Review by driskull — APPROVED** [review]
LGTM.

Lets get another reviewer since I contributed.

**Comment by asangma** [comment]
Thanks @anveshmekala!
It's a better situation.

**Review by asangma — APPROVED** [review]
Whoop whoop!

File: src/components/calcite-dropdown/calcite-dropdown.e2e.ts [hunk 2]
```diff
@@ -1031,4 +1031,35 @@ describe("calcite-dropdown", () => {
       })
     ).toBe(false);
   });
+
+  it("dropdown wrapper shouldnt' have height when filter results empty and combined with a PickList in Panel  #3048", async () => {
```
Comment (asangma) [inline src/components/calcite-dropdown/calcite-dropdown.e2e.ts:1035]: typo: "shouldn't"

File: src/components/calcite-dropdown/calcite-dropdown.tsx [hunk 3]
```diff
@@ -316,6 +318,8 @@ export class CalciteDropdown {
 
   mutationObserver = createObserver("mutation", () => this.updateItems());
 
+  resizeObserver = createObserver("resize", () => this.setMaxScrollerHeight());
```
Comment (driskull) [inline src/components/calcite-dropdown/calcite-dropdown.tsx:321]: @anveshmekala we might just be able to call `reposition()` along with `this.setMaxScrollerHeight()` in order to fix https://github.com/Esri/calcite-components/issues/3234

Comment (driskull) [reply src/components/calcite-dropdown/calcite-dropdown.tsx:321]: I verified that calling `this.reposition();` inside `setMaxScrollerHeight` fixes the issue.

@anveshmekala

**Pull request merged**